### ISSUE TITLE
Textual: equality syntax sugar

### ISSUE THREAD

**Issue by dcz-self** [fatho/logru#24]
Playing with the REPL example, I noticed that the equals sign is not recognized. In my usage of Prolog, it's rather common:

```
?- A = config(_, _).
Failed to parse: ParseError { span: 2..3, kind: UnrecognizedToken }
?- :define eq(A, A).
Defined!
?- eq(A, config(_, _)).
Found solution:
  A = config(_1, _2)
```

For my application, it would be rather important to be able to leverage general Prolog knowledge - a Prolog-like solver is already unfamiliar territory for the intended user of my project (finding [camera](https://codeberg.org/libobscura/libobscura) configs).

It's an infix operator, as opposed to what logru supports now, and it would translate to an actual predicate, while as far as I can tell, there is no standard library in logru at all.

Would you be interested if I tried to implement it? What's the best place?

**Comment by fatho** [comment]
> Would you be interested if I tried to implement it?

Definitely, if you would like to give it a try, that'd be much appreciated.

> What's the best place?

I was already thinking of introducing proper `,` and `;` builtins for conjunctions and disjunctions, rather than the current way of storing implicit conjunctions as `Vec<TermId>`s in `Rule` and `Query`, and also parsing parenthesized terms, so that you can have something like `foo, bar, (baz ; bla)` as a term.

A built-in equality would fit in well there, and would be a good way to start.
I'd probably implement it as a proper term type, just like `Cut` is already, so
- adding a `Eq` constructor to `ast` and `term_arena` level `Term`s
- handling `Eq` terms as a special case in `SolutionIter::step` by unifying both sides
- recognizing `=` in the lexer
- parsing `Eq` terms as infix operator with the correct associativity (not sure how it's parsed in standard prologs).

The parser would later need to be extended to support proper precedence parsing so that it all interacts correctly with `,` and the planned `;`. For that, I planned to take some inspiration of a parser I've written earlier for another project: https://github.com/fatho/syn-txt/blob/1b6497caa147c43b4bcfe8c497566ee62b709268/syntxt-lang/src/parser.rs#L328-L408

**Comment by dcz-self** [comment]
Thanks for the intro. I'm not making any promises right now, this only becomes urgent when people actually start using what I made.
